### PR TITLE
Improve frontend dockerfile

### PR DESCRIPTION
**Pull request by titilambert**
## Improve frontend dockerfile

I added `yq` to the Frontend dockerfile to be able to read configuration from the yq file.
This permits to handle configuration with array structures (like homepage.clocks....)

#### :heavy_check_mark: Checklist

- [ ] Added or updated documentation

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4535-nwodz2bjxx3lgnq3d9oqv09euz2jwmzg.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602a89ad65f8172d20f0ba76

File: contrib/docker/frontend-with-nginx/docker/run.sh [hunk 1]
```diff
@@ -10,14 +10,27 @@ envsubst '$PORT' < /etc/nginx/conf.d/default.conf.template > /etc/nginx/conf.d/d
 
 # Inject runtime config into the client
 function inject_config() {
+  # Read runtime config from yaml file (it could be the backend one)
```
Comment (Rugvip) [inline contrib/docker/frontend-with-nginx/docker/run.sh:13]: Absolutely could not be the same as the backend one 😅 , it won't be filtered for secrets, so any secrets in the app config would be sent straight on to the frontend.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thanks!

Not entirely opposed to the idea of being able to load in config from a file in addition to env, but need to take a bit more caution imo. The app-config.yaml files require quite a bit of processing to be used properly, both for replacing things like `$env` keys and other substitutions, but also filter for visibility in the frontend using the config schema.

We could add something like this, but I think we should make it very clear that it will be used in unprocessed form and sent directly to the frontend, and should not be shared with backend config. Perhaps by simply calling the file `frontend-config.yaml`?

**Comment by Rugvip** [comment]
@titilambert what's your thoughts on the way forward? Do you want to keep the logic and update the docs a bit?

**Comment by freben** [comment]
Hi! We're closing this one for now. Let us know if you think that was in error.

**Pull request closed without merging**